From 7c780f8a67d42056100b93440fc0a6f97b04eede Mon Sep 17 00:00:00 2001
From: Oleksandr Tyshchenko <oleksandr_tyshchenko@epam.com>
Date: Tue, 2 Feb 2021 18:42:36 +0200
Subject: [PATCH 2/3] tools: Produce two cert_header_sa6 images

An already existed image is used to boot SoC from the Hyperflash,
while the newly created image (cert_header_sa6_emmc) is needed to boot
SoC from EMMC boot partition 1.
For producing suitable for "eMMC boot" image we have to build sa6.c
source file again, but with RCAR_SA6_TYPE flag being forced to 1.

Signed-off-by: Oleksandr Tyshchenko <oleksandr_tyshchenko@epam.com>
Signed-off-by: Valerii Chubar <valerii_chubar@epam.com>
---
 tools/renesas/rcar_layout_create/makefile | 22 +++++++++++++++++++++-
 1 file changed, 21 insertions(+), 1 deletion(-)

diff --git a/tools/renesas/rcar_layout_create/makefile b/tools/renesas/rcar_layout_create/makefile
index 4bfea7dc6..0577e396a 100644
--- a/tools/renesas/rcar_layout_create/makefile
+++ b/tools/renesas/rcar_layout_create/makefile
@@ -11,18 +11,26 @@
 #output file name
 FILE_NAME_SA0   = bootparam_sa0
 FILE_NAME_SA6   = cert_header_sa6
+FILE_NAME_SA6_EMMC = cert_header_sa6_emmc
 
 OUTPUT_FILE_SA0 = $(FILE_NAME_SA0).elf
 OUTPUT_FILE_SA6 = $(FILE_NAME_SA6).elf
+OUTPUT_FILE_SA6_EMMC = $(FILE_NAME_SA6_EMMC).elf
 
 #object file name
 OBJ_FILE_SA0 =	sa0.o
 OBJ_FILE_SA6 =	sa6.o
+OBJ_FILE_SA6_EMMC =	sa6_emmc.o
 
 #linker script name
 MEMORY_DEF_SA0 = sa0.ld.S
 MEMORY_DEF_SA6 = sa6.ld.S
 
+# source file name
+SOURCE_FILE_SA6 =	sa6.c
+# Explicitly set RCAR_SA6_TYPE flag
+CFLAGS_SA6_EMMC = `echo "${CFLAGS}" | sed -e 's/-DRCAR_SA6_TYPE=0/-DRCAR_SA6_TYPE=1/'`
+
 ###################################################
 # Convenience function for adding build definitions
 # $(eval $(call add_define,FOO)) will have:
@@ -89,7 +97,7 @@ CL = rm -f
 # command
 
 .PHONY: all
-all: $(OUTPUT_FILE_SA0) $(OUTPUT_FILE_SA6)
+all: $(OUTPUT_FILE_SA0) $(OUTPUT_FILE_SA6) $(OUTPUT_FILE_SA6_EMMC)
 ###################################################
 # Linker
 ###################################################
@@ -111,10 +119,22 @@ $(OUTPUT_FILE_SA6) : $(MEMORY_DEF_SA6) $(OBJ_FILE_SA6)
 	$(objcopy) -O srec --adjust-vma=$(RCAR_VMA_ADJUST_ADDR) --srec-forceS3  $(OUTPUT_FILE_SA6) $(FILE_NAME_SA6).srec
 	$(objcopy) -O binary --adjust-vma=$(RCAR_VMA_ADJUST_ADDR) --srec-forceS3  $(OUTPUT_FILE_SA6) $(FILE_NAME_SA6).bin
 
+$(OUTPUT_FILE_SA6_EMMC) : $(MEMORY_DEF_SA6) $(OBJ_FILE_SA6_EMMC)
+	$(LD) $(OBJ_FILE_SA6_EMMC)      \
+	-T $(MEMORY_DEF_SA6)            \
+	-o $(OUTPUT_FILE_SA6_EMMC)      \
+	-Map $(FILE_NAME_SA6_EMMC).map  \
+
+	$(objcopy) -O srec --adjust-vma=0xE6320000 --srec-forceS3  $(OUTPUT_FILE_SA6_EMMC) $(FILE_NAME_SA6_EMMC).srec
+	$(objcopy) -O binary --adjust-vma=0xE6320000 --srec-forceS3  $(OUTPUT_FILE_SA6_EMMC) $(FILE_NAME_SA6_EMMC).bin
+
 ###################################################
 # Compile
 ###################################################
 
+$(OBJ_FILE_SA6_EMMC):$(SOURCE_FILE_SA6)
+	${CC} -c ${CFLAGS_SA6_EMMC} ${CPPFLAGS} $< -o $@
+
 %.o:../%.c
 	$(CC) -c -I $< -o $@
 
-- 
2.17.1

